File: go/generate-parentheses/generate-parentheses.go
package main

import(
	"fmt"
)

func generateParenthesis(n int)[]string{

	var solve func(open, close int, temp string)
    ans := []string{}

    solve = func(open, close int, temp string) {
        if len(temp) == 2*n {
            ans = append(ans, temp)
            return
        }

        if open > 0 {
            solve(open-1, close, temp+"(")
        }
        if close > open {
            solve(open, close-1, temp+")")
        }
    }

    solve(n, n, "")
    return ans
}

func main(){

	n := 3

	fmt.Printf("With %d pairs of parentheses all the well formed couples are: %v", n, generateParenthesis(n))
}

File: js/generate-parentheses/generate-parentheses.js
function generateParenthesis(n){

    const res = []

    function dfs(openP, closeP, s) {
        if (openP === closeP && openP + closeP === n * 2) {
            res.push(s)
            return
        }

        if (openP < n) {
            dfs(openP + 1, closeP, s + "(")
        }

        if (closeP < openP) {
            dfs(openP, closeP + 1, s + ")")
        }
    }

    dfs(0, 0, "")

    return res   
}

let n = 3 
console.log(`With ${n} pairs of parentheses all the well formed couples are:`, generateParenthesis(n))